### ISSUE TITLE
Meson build error on Debian 9 Stretch 

### ISSUE THREAD

**Issue by second-story** [gnumdk/eolie#255]
### Environment
- Eolie version: 0.9.2
- GNOME version: 3.22.2
- GTK+ version: 3.22.11-1
- GNU/Linux distribution: Debian 9 Stretch

I cloned Eolie and tried to build it for my Debian 9 system, but Meson errors out right after executing `meson build`.  Meson gives the following error message:

```
The Meson build system
Version: 0.42.0
Source dir: /home/ricky/bin/opt/eolie
Build dir: /home/ricky/bin/opt/eolie/build
Build type: native build
Project name: eolie
Native C compiler: cc (gcc 6.3.0)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Message: Looking for dependencies
Message: Found python3 binary
Found pkg-config: /usr/bin/pkg-config (0.29)
Native dependency glib-2.0 found: YES 2.47.5
Native dependency gobject-introspection-1.0 found: YES 1.50.0

Meson encountered an error in file meson.build, line 18, column 0:
Could not generate cargs for gtk+-3.0:
```

Which is odd, since my gtk3 meets the requirements.  Specifically, I have libgtk-3-0:amd64, libgtk-3-bin, libgtk-3-common, and libgtk-3-dev:amd64 installed to my system, and each is version 3.22.11-1.

Am I missing a dependency or is this something else?

Your work on Lollypop is amazing, btw.  Really looking forward to using this program too.

**Comment by gnumdk** [comment]
Have you tried with Lollypop?

**Comment by second-story** [comment]
I actually did try just before attempting to build this, but was given the exact same error (my Lollypop still runs though because it was built months earlier).  Whatever issue is preventing the cargs generation is identical between the two programs.

**Comment by gnumdk** [comment]
Output of:
$ pkg-config gtk+-3.0 --libs

**Comment by second-story** [comment]
Here's the output: 

`Package 'gtk+-3.0' requires 'gio-2.0 >= 2.49.4' but version of gio-2.0 is 2.47.5`

I'm a little confused how my 1.50.0-1+b1 version of gio-2.0 (listed as gir1.2-glib-2.0 on Debian) is so vastly different than the number required by pkg-config, especially when the most current Debian version of gir1.2-glib-2.0 is 1.54.0-2.  

I have a feeling that compiling gio from https://git.gnome.org//browse/glib/ is in order, unless I'm misunderstanding something about pkg-config's output.

**Comment by gnumdk** [comment]
Looks like your configuration is broken... Is it a real Debian 9 or testing?

**Comment by gnumdk** [comment]
prefix=/usr
exec_prefix=${prefix}
libdir=${prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include

giomoduledir=${libdir}/gio/modules
glib_compile_schemas=${prefix}/lib/x86_64-linux-gnu/glib-2.0/glib-compile-schemas
glib_compile_resources=${prefix}/lib/x86_64-linux-gnu/glib-2.0/glib-compile-resources
gdbus_codegen=gdbus-codegen

Name: GIO
Description: glib I/O library
Version: 2.50.3
Requires: glib-2.0 gobject-2.0
Requires.private: gmodule-no-export-2.0
Libs: -L${libdir} -lgio-2.0
Libs.private: -lz -lresolv  -lselinux  
Cflags:

**Comment by gnumdk** [comment]
This is gio from Debian 9, if you upgraded from Debian 8, something is wrong...

# apt-get dist-upgrade?

**Comment by second-story** [comment]
That could definitely be the case.  I've been upgrading this thing since 2013, when Debian 7 was freshly released, and it's accumulated all kinds of cruft from back when I knew nothing about Debian.  

I was actually going to do a reinstall to fix some partitioning mistakes from back then, and will try to rebuild this and Lollypop on the new system when it's ready.

Out of curiosity, what command did you run to get the output from https://github.com/gnumdk/eolie/issues/255#issuecomment-330822065?

**Comment by gnumdk** [comment]
I searched for gio2
-2.0.pc file at packages.debian.org

**Comment by gnumdk** [comment]
But don't reinstall, apt-get dist-upgrade should do the job.

**Comment by second-story** [comment]
lol at just searching packages.debian.org.  I thought you executed pkg-config with a special flag.  

I tried performing a dist-upgrade before my last comment, but my system reported then (and now) that everything was up to date.  Even trying to dist-upgrade through stretch-backports produced nothing.  I had a feeling it couldn't be that easy.

I was going to reinstall anyway to smash all the old setup issues I mentioned on https://github.com/gnumdk/eolie/issues/255#issuecomment-331020147, so it's really no problem for me.  It'll probably be another week before I do that, though, so this issue is probably safe to close.  I'll open another issue if Eolie still fails to build after a reinstall, since my situation would then more accurately reflect other Debian users.  Until then, thanks for the help!
